### DISCUSSION TITLE
unexpected keyword argument 'token'

### DISCUSSION THREAD

**Discussion by saiaprameya** [PromtEngineer/localGPT#721] [General]
when running python ingest.py getting the following traceback:

Traceback (most recent call last):
  File "/home/test/doc-inferencing/ingest.py", line 181, in <module>
    main()
  File "/home/test/.conda/envs/localGPT/lib/python3.10/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
  File "/home/test/.conda/envs/localGPT/lib/python3.10/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
  File "/home/test/.conda/envs/localGPT/lib/python3.10/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/test/.conda/envs/localGPT/lib/python3.10/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/home/test/doc-inferencing/ingest.py", line 157, in main
    embeddings = HuggingFaceInstructEmbeddings(
  File "/home/test/.conda/envs/localGPT/lib/python3.10/site-packages/langchain/embeddings/huggingface.py", line 148, in __init__
    self.client = INSTRUCTOR(
  File "/home/test/.conda/envs/localGPT/lib/python3.10/site-packages/sentence_transformers/SentenceTransformer.py", line 194, in __init__
    modules = self._load_sbert_model(
TypeError: INSTRUCTOR._load_sbert_model() got an unexpected keyword argument 'toke

Can you please let me know if i am missing anything here. Have followed the procedure as described.

pip freeze output:

absl-py==2.1.0
accelerate==0.26.1
aiohttp==3.9.3
aiosignal==1.3.1
altair==5.2.0
antlr4-python3-runtime==4.9.3
anyio==4.2.0
async-timeout==4.0.3
attributedict==0.3.0
attrs==23.2.0
auto-gptq==0.2.2
autoawq==0.1.8
backoff==2.2.1
beautifulsoup4==4.12.3
bitsandbytes==0.42.0
blessings==1.7
blinker==1.7.0
cachetools==5.3.2
certifi==2023.11.17
cffi==1.16.0
chardet==5.2.0
charset-normalizer==3.3.2
chroma-hnswlib==0.7.2
chromadb==0.4.6
click==8.1.7
codecov==2.1.13
colorama==0.4.6
coloredlogs==15.0.1
colour-runner==0.1.1
contourpy==1.2.0
coverage==7.4.1
cryptography==42.0.1
cycler==0.12.1
dataclasses-json==0.5.14
DataProperty==1.0.1
datasets==2.16.1
deepdiff==6.7.1
Deprecated==1.2.14
dill==0.3.7
diskcache==5.6.3
distlib==0.3.8
docx2txt==0.8
effdet==0.4.1
emoji==2.10.0
et-xmlfile==1.1.0
evaluate==0.4.1
exceptiongroup==1.2.0
faiss-cpu==1.7.4
fastapi==0.99.1
filelock==3.13.1
filetype==1.2.0
Flask==3.0.1
flatbuffers==23.5.26
fonttools==4.47.2
frozenlist==1.4.1
fsspec==2023.10.0
gitdb==4.0.11
GitPython==3.1.41
greenlet==3.0.3
h11==0.14.0
httptools==0.6.1
huggingface-hub==0.20.3
humanfriendly==10.0
idna==3.6
importlib-metadata==7.0.1
importlib-resources==6.1.1
inspecta==0.1.3
InstructorEmbedding==1.0.1
iopath==0.1.10
itsdangerous==2.1.2
Jinja2==3.1.3
joblib==1.3.2
jsonlines==4.0.0
jsonschema==4.21.1
jsonschema-specifications==2023.12.1
kiwisolver==1.4.5
langchain==0.0.267
langdetect==1.0.9
langsmith==0.0.84
layoutparser==0.3.4
llama_cpp_python==0.1.83
lm_eval==0.4.0
lxml==5.1.0
markdown-it-py==3.0.0
MarkupSafe==2.1.4
marshmallow==3.20.2
matplotlib==3.8.2
mbstrdecoder==1.1.3
mdurl==0.1.2
monotonic==1.6
mpmath==1.3.0
multidict==6.0.4
multiprocess==0.70.15
mypy-extensions==1.0.0
networkx==3.2.1
nltk==3.8.1
numexpr==2.9.0
numpy==1.26.3
nvidia-cublas-cu12==12.1.3.1
nvidia-cuda-cupti-cu12==12.1.105
nvidia-cuda-nvrtc-cu12==12.1.105
nvidia-cuda-runtime-cu12==12.1.105
nvidia-cudnn-cu12==8.9.2.26
nvidia-cufft-cu12==11.0.2.54
nvidia-curand-cu12==10.3.2.106
nvidia-cusolver-cu12==11.4.5.107
nvidia-cusparse-cu12==12.1.0.106
nvidia-nccl-cu12==2.18.1
nvidia-nvjitlink-cu12==12.3.101
nvidia-nvtx-cu12==12.1.105
omegaconf==2.3.0
onnx==1.15.0
onnxruntime==1.15.1
openapi-schema-pydantic==1.2.4
opencv-python==4.9.0.80
openpyxl==3.1.2
ordered-set==4.1.0
overrides==7.7.0
packaging==23.2
pandas==2.2.0
pathvalidate==3.2.0
pdf2image==1.17.0
pdfminer.six==20221105
pdfplumber==0.10.3
peft==0.7.1
pikepdf==8.11.2
pillow==10.2.0
platformdirs==4.1.0
pluggy==1.4.0
portalocker==2.8.2
posthog==3.3.3
protobuf==3.20.2
psutil==5.9.8
pulsar-client==3.4.0
pyarrow==15.0.0
pyarrow-hotfix==0.6
pybind11==2.11.1
pycocotools==2.0.7
pycparser==2.21
pydantic==1.10.14
pydeck==0.8.1b0
Pygments==2.17.2
pyparsing==3.1.1
pypdf==4.0.1
pypdfium2==4.26.0
PyPika==0.48.9
pyproject-api==1.6.1
pytablewriter==1.2.0
pytesseract==0.3.10
python-dateutil==2.8.2
python-dotenv==1.0.1
python-iso639==2024.1.2
python-magic==0.4.27
python-multipart==0.0.6
pytz==2023.4
PyYAML==6.0.1
rapidfuzz==3.6.1
referencing==0.33.0
regex==2023.12.25
requests==2.31.0
responses==0.18.0
rich==13.7.0
rootpath==0.1.1
rouge==1.0.1
rouge-score==0.1.2
rpds-py==0.17.1
sacrebleu==2.4.0
safetensors==0.4.2
scikit-learn==1.4.0
scipy==1.12.0
sentence-transformers==2.3.0
sentencepiece==0.1.99
six==1.16.0
smmap==5.0.1
sniffio==1.3.0
soupsieve==2.5
SQLAlchemy==2.0.25
sqlitedict==2.1.0
starlette==0.27.0
streamlit==1.30.0
streamlit-extras==0.0.9
sympy==1.12
tabledata==1.3.3
tabulate==0.9.0
tcolorpy==0.1.4
tenacity==8.2.3
termcolor==2.4.0
texttable==1.7.0
threadpoolctl==3.2.0
timm==0.9.12
tokenizers==0.15.1
toml==0.10.2
tomli==2.0.1
toolz==0.12.1
torch==2.1.2
torchvision==0.16.2
tornado==6.4
tox==4.12.1
tqdm==4.66.1
tqdm-multiprocess==0.0.11
transformers==4.37.2
triton==2.1.0
typepy==1.3.2
typing-inspect==0.9.0
typing_extensions==4.9.0
tzdata==2023.4
tzlocal==5.2
unstructured==0.11.2
unstructured-inference==0.7.15
unstructured.pytesseract==0.3.12
urllib3==1.26.6
uvicorn==0.27.0.post1
uvloop==0.19.0
validators==0.22.0
virtualenv==20.25.0
watchdog==3.0.0
watchfiles==0.21.0
websockets==12.0
Werkzeug==3.0.1
wrapt==1.16.0
xxhash==3.4.1
yarl==1.9.4
zipp==3.17.0
zstandard==0.22.0

**Comment by aegaji1** [comment]
I found this helpful - seems it was updated recently.
downgrade to langchain==0.1.2 sentence_transformers==2.2.2
https://stackoverflow.com/questions/77903917/typeerror-for-while-using-instruct-embeddings-on-huggingface

**Reply by Kamruzzamansust** [reply]
first i uninsall langchain and sentence_transformers and install these langchain==0.1.2 sentence_transformers==2.2.2 . but still get the same error

**Reply by ApJML** [reply]
Only reinstalling sentence_transformers==2.2.2 worked. Thanks.

**Comment by fukuro-kun** [comment]
The following commands fixed the issue for me, I did not need to change langchain version at all.

```
pip uninstall sentence-transformers
pip install sentence-transformers==2.2.2
```

**Comment by dmsolutionz** [comment]
They resolved this in the latest main thankfully - I was going mad yesterday 😂

**Comment by saiaprameya** [comment]
Thanks all.
As suggested this got resolved:
pip uninstall sentence-transformers
pip install sentence-transformers==2.2.2

**Comment by AhmedHakim0401** [comment]
I got the same error with sentence-transformers==2.2.2
